Add document about canonical transcript (#774)
genome-nexus/genome-nexus: docs/Select-Canonical-Transcript.md

# How Genome Nexus selects a canonical transcript

## **Background**
When a variant is sent to Genome Nexus, Genome Nexus calls **VEP** (Variant Effect Predictor) to get a range of annotations associated with the variant. Most of these annotations are associated with a particular transcript. For example. [7:g.55249071C>T](https://www.genomenexus.org/variant/7:g.55249071C%3ET) is `c.2369C>T / p.T790M` in `ENST00000275493.2`, but `c.2210C>T / p.T737M` for transcript `ENST00000454757.2`.

Genome Nexus aims to select one transcript (the canonical transcript) from this list of transcripts to have a default most relevant to oncology. The selection process is guided by a curated list of paired gene and transcript preferences (e.g., `EGFR` and `ENST00000275493.2`) and additional prioritization logic that emphasizes mutations with a high impact (e.g., in coding regions and/or targeting cancer-related genes). The curated list of gene and transcripts is called the isoform override list.

### Available isoform override sources
The isoform override source can be defined in `isoformOverrideSource` field, options are:
- `mskcc`

> - `uniprot` (default)

Reviewer: should we make `mskcc` the default? As it is in cBioPortal

        - Description: This file contains the GRCh37 mskcc isoform canonical transcript list. If a transcript is not present in the OncoKB canonical transcript list, this file is used as a fallback.

    - GRCh38: 
      - [isoform_overrides_oncokb_grch38.txt](https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_at_mskcc_grch38.txt)
        - Source: [OncoKB curated genes](https://www.oncokb.org/api/v1/utils/allCuratedGenes.txt?includeEvidence=false)
        - Description: This file contains the GRCh38 OncoKB isoform canonical transcript list. It is the primary choice when `mskcc` is selected as the isoform override source.
      - [isoform_overrides_at_mskcc_grch38.txt](https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_genome_nexus_grch38.txt)
        - Source: [MSKCC VCF2MAF Isoform Overrides](https://github.com/mskcc/vcf2maf/blob/main/data/isoform_overrides_at_mskcc_grch38)
        - Description: This file contains the GRCh38 MSKCC isoform canonical transcript list. If a transcript is not present in the OncoKB canonical transcript list, this file is used as a fallback.

> - `uniprot`:

Reviewer: I think it's good to mention the caveats around `uniprot` as outlined here: https://docs.cbioportal.org/mutation-data-transcript-annotation/#transcript-assignment. The name `uniprot` seems to imply that it comes from uniprot somehow so can be confusing to a user

# How Genome Nexus selects a canonical transcript

## **Background**

> When a variant is sent to Genome Nexus, Genome Nexus calls **VEP** (Variant Effect Predictor)to get a range of annotations associated with the variant, including a list of potential transcripts.

Reviewer: ```suggestion
When a variant is sent to Genome Nexus, Genome Nexus calls **VEP** (Variant Effect Predictor) to get a range of annotations associated with the variant. Most of these annotations are associated with a particular transcript. For example. [7:g.55249071C>T](https://www.genomenexus.org/variant/7:g.55249071C%3ET) is `c.2369C>T / p.T790M` in `ENST00000275493.2`, but `c.2210C>T / p.T737M` for transcript `ENST00000454757.2`)
```

> Genome Nexus then uses a set of predefined criteria to select the canonical transcript from this list of transcripts. The selection process is guided by the user-selected isoform override source and a prioritization logic.

Reviewer: ```suggestion
Genome Nexus aims to select one transcript (the canonical transcript) from this list of transcripts to have a default most relevant to oncology. The selection process is guided by a curated list of paired gene and transcript preferences (e.g., `EGFR` and `ENST00000275493.2`) and additional prioritization logic that emphasizes mutations with a high impact (e.g., in coding regions and/or targeting cancer-related genes).
```

The canonical transcript is in the `transcriptConsequenceSummary` field, under the `annotation_summary` section of the response. To ensure the canonical transcript is returned, the `annotation_summary` field must be requested in the query sent to Genome Nexus.

Example response structure:
```
"annotation_summary" {
    "transcriptConsequenceSummary" {
        ...
    }
}

> ```

Reviewer: this is quite technical, I'm wondering if it's better to split this out in separate sections, e.g. "command-line client", "REST API", "website" etc. The basics of the prioritization logic is useful for all users, but some details are only relevant for a subset of downstream users

and 
https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/grch38_ensembl92/export/ensembl_biomart_canonical_transcripts_per_hgnc.txt

The original isoform override files can be found here:
- `mskcc`:
    - GRCh37: https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_oncokb_grch37.txt 
    and 
    https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_oncokb_grch37.txt
    - GRCh38: https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_at_mskcc_grch38.txt
    and
    https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_oncokb_grch38.txt
- `uniprot`:
    - https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_uniprot.txt
- `genome_nexus`:
    - GRCh37: https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_genome_nexus_grch37.txt
    - GRCh38: https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/common_input/isoform_overrides_genome_nexus_grch38.txt

> 

Reviewer: would be good to clarify for each of these where they originated from and why there were introduced

---

## Overview
![How Genome Nexus selects canonical transcript (1)](https://github.com/user-attachments/assets/b0df1faa-9819-462a-aa16-0c3e61db08f0)

### **Step 1: Send a variant**

> - When a variant is sent to Genome Nexus, there are some other parameters sending in the query as well, such `isoformOverrideSource` and `fields`.

Reviewer: ```suggestion
- When a variant is sent to Genome Nexus, there are some other parameters sent in the query as well, such `isoformOverrideSource` and `fields`.
```

This works but it is pretty technical. Wondering if it's better to explain at a high level for all users and then have additional details using e.g. some [markdown Note](https://www.freecodecamp.org/news/how-to-create-notice-blocks-in-markdown/) for technical users
